src/game-logic/game-loop.ts

```diff
@@ -30,28 +30,34 @@ const nextState = (currentState: GameState, inputs: InputStatus): GameState => {
 
   switch (currentState.screen) {
     case 'in-game':
-      let state = currentState
-      state = handlePlayerInputs(state, inputs, keysPressed, keysReleased)
-      state = updateAttacks(state)
-      state = nextPhysicsState(state)
+      let gameState = currentState
+      gameState = handlePlayerInputs(gameState, inputs, keysPressed, keysReleased)
+      gameState = updateAttacks(gameState)
+      gameState = nextPhysicsState(gameState)
 
-      if (isGameOver(state)) {
-        return gameOverState(state.players)
+      if (isGameOver(gameState)) {
+        return gameOverState(gameState.players)
```

Comment:
Why rename this if the others are still called `state`?

Reply:
Cannot re-declare block-scoped variables. Every currentState.screen case needs to refer to the state in their own way. Perhaps the correct approach here would be to set `let state = currentState` outside the switch statement. What do you think?

Reply:
That's actually a very good point. The correct way would be to use `let state = { ...currentState }` on top level, and change all references to `currentState` to use `state`, so we can get correct type inference from the switch statement. I implemented this on dev, so please merge and let me know if you run into any issues.